robusta_krr/core/runner.py

```diff
@@ -143,29 +143,27 @@ async def _calculate_object_recommendations(self, object: K8sObjectData) -> RunR
             step=self._strategy.settings.timeframe_timedelta,
         )
 
-        self.__progressbar.progress()
+        self.debug(f"Calculating recommendations for {object} with {len(metrics)} metrics")
 
         # NOTE: We run this in a threadpool as the strategy calculation might be CPU intensive
         # But keep in mind that numpy calcluations will not block the GIL
         loop = asyncio.get_running_loop()
         result = await loop.run_in_executor(self._executor, self._strategy.run, metrics, object)
         return self._format_result(result)
 
-    async def _gather_objects_recommendations(self, objects: list[K8sObjectData]) -> list[ResourceAllocations]:
-        recommendations: list[RunResult] = await asyncio.gather(
-            *[self._calculate_object_recommendations(object) for object in objects]
-        )
+    async def _gather_object_allocations(self, k8s_object: K8sObjectData) -> ResourceScan:
+        recommendation = await self._calculate_object_recommendations(k8s_object)
 
-        return [
-            (
-                ResourceAllocations(
-                    requests={resource: recommendation[resource].request for resource in ResourceType},
-                    limits={resource: recommendation[resource].limit for resource in ResourceType},
-                    info={resource: recommendation[resource].info for resource in ResourceType},
-                )
-            )
-            for recommendation in recommendations
-        ]
+        self.__progressbar.progress()
+
+        return ResourceScan.calculate(
```

Comment:
out of curiosity does calculating here improve krr memory?

Reply:
This is for a refactoring

Reply:
This is only for a refactoring